opslevel/resource_opslevel_service_repository.go

```diff
@@ -203,6 +203,10 @@ func (r *ServiceRepositoryResource) Read(ctx context.Context, req resource.ReadR
 		service, err = r.client.GetServiceWithAlias(currentStateModel.ServiceAlias.ValueString())
 	}
 	if err != nil {
+		if (service == nil || service.Id == "") && opslevel.IsOpsLevelApiError(err) {
```

Comment:
same issues - only `RemoveResource` if the `ServiceRepository` is not found on the service - error if the service isn't found.